core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -260,35 +262,37 @@ constructor(
         currentSettings
             .filterNotNull()
             .map { currentCameraSettings ->
-                transientSettings.value = TransientSessionSettings(
-                    isAudioMuted = currentCameraSettings.audioMuted,
-                    deviceRotation = currentCameraSettings.deviceRotation,
-                    flashMode = currentCameraSettings.flashMode,
-                    zoomScale = currentCameraSettings.zoomScale
-                )
-
-                val cameraConstraints = checkNotNull(
-                    systemConstraints.forCurrentLens(currentCameraSettings)
-                ) {
-                    "Could not retrieve constraints for ${currentCameraSettings.cameraLensFacing}"
-                }
-
-                val resolvedStabilizationMode = resolveStabilizationMode(
-                    stabilizationMode = currentCameraSettings.stabilizationMode,
-                    targetFrameRate = currentCameraSettings.targetFrameRate,
-                    supportedStabilizationModes = cameraConstraints.supportedStabilizationModes,
-                    concurrentCameraMode = currentCameraSettings.concurrentCameraMode
-                )
-
                 when (currentCameraSettings.concurrentCameraMode) {
                     ConcurrentCameraMode.OFF -> {
-                        val cameraSelector = when (currentCameraSettings.cameraLensFacing) {
-                            LensFacing.FRONT -> CameraSelector.DEFAULT_FRONT_CAMERA
-                            LensFacing.BACK -> CameraSelector.DEFAULT_BACK_CAMERA
+                        cameraSelector.update {
+                            when (currentCameraSettings.cameraLensFacing) {
+                                LensFacing.FRONT -> CameraSelector.DEFAULT_FRONT_CAMERA
+                                LensFacing.BACK -> CameraSelector.DEFAULT_BACK_CAMERA
+                            }
+                        }
+
+                        transientSettings.value = TransientSessionSettings(
+                            isAudioMuted = currentCameraSettings.audioMuted,
+                            deviceRotation = currentCameraSettings.deviceRotation,
+                            flashMode = currentCameraSettings.flashMode,
+                            zoomScale = currentCameraSettings.zoomScale
+                        )
```

Comment:
Moving this here seems to have made it so we can no longer zoom in concurrent camera mode. I assume it might also do the same for muting audio and flash.